apps/designer/app/designer/features/style-panel/sections/spacing/spacing.tsx

```diff
@@ -111,61 +120,87 @@ const styles = {
 
 type TextFieldProps = {
   property: StyleProperty;
-  value: string | undefined;
+  value?: UnitValue;
   onEnter: (value: string) => void;
 };
 
+const initialValue: UnitValue = {
+  value: 0,
+  unit: "px",
+  type: "unit",
+};
+
+const slideUpAndFade = keyframes({
+  "0%": { opacity: 0, transform: "scale(0)" },
+  "100%": { opacity: 1, transform: "scale(1)" },
+});
+
 const TextField = ({ property, value, onEnter }: TextFieldProps) => {
-  const [currentValue, setCurrentValue] = useState<string>(value ?? "");
+  const [currentValue, setCurrentValue] = useState<UnitValue>(
+    value ?? initialValue
+  );
   const isFromCurrentBreakpoint = useIsFromCurrentBreakpoint(property);
 
   useEffect(() => {
-    setCurrentValue(value ?? "");
+    setCurrentValue(value ?? initialValue);
   }, [value]);
 
+  const [isOpen, setIsOpen] = useState(false);
+
   return (
-    <Box
-      as="input"
-      name={property}
-      aria-label={`${property} edit`}
-      value={currentValue}
-      onKeyDown={(event: React.KeyboardEvent<HTMLInputElement>) => {
-        if (event.key === "Enter") {
-          onEnter(currentValue);
-        }
-      }}
-      onChange={(event: React.ChangeEvent<HTMLInputElement>) => {
-        setCurrentValue(event.target.value);
-      }}
-      css={
-        isFromCurrentBreakpoint
-          ? { ...styles.input, ...styles.inputFromCurrentBreakpoint }
-          : styles.input
-      }
-    />
+    <Popover modal open={isOpen} onOpenChange={setIsOpen}>
+      <PopoverTrigger asChild>
+        <Box
+          aria-label={`${property} edit`}
+          css={{
+            ...(isFromCurrentBreakpoint
+              ? { ...styles.input, ...styles.inputFromCurrentBreakpoint }
+              : styles.input),
+          }}
+        >
+          {toValue(currentValue)}
+        </Box>
+      </PopoverTrigger>
+      <PopoverPortal>
+        <PopoverContent
+          hideArrow
+          css={{
+            background: "none",
+            boxShadow: "none",
+            borderRadius: 0,
+            minWidth: 80,
+            width: 80,
+            padding: 2,
+            animationDuration: "200ms",
+            animationTimingFunction: "cubic-bezier(0.16, 1, 0.3, 1)",
+            willChange: "transform, opacity",
```

Comment:
willChange  usually has no effect or even degrades perf.  Also doc `Warning: will-change is intended to be used as a last resort, in order to try to deal with existing performance problems. It should not be used to anticipate performance problems.`

Here depends on Popover implementation, as setting it with element creation usually has no effect. Should be set before animation begins. Anway somekind of magic.